src/server/system_services/bucket_server.js

```diff
@@ -1933,7 +1932,7 @@ function validate_replication(req, db_repl_rules) {
         }
 
         // validation for bidirectional replication - blocking bidirectional replication only for matching prefixes
-        if (destination_bucket.unwrap() !== req.rpc_params.name.unwrap()) {
+        if (destination_bucket.unwrap() !== req.rpc_params.name.unwrap() && src_bucket) {
```

Comment:
@dannyzaken @liranmauda This condition here is specific to obc, as for obc `validate_replication()` is called two times one before the creation of src_bucket and one after the src_bucket creation inside `put_bucket_replication()`. Because of this condition check only execute if src_bucket is already present/created.

Thats the best way I found here, pls feel free to provide more suggestions.

Reply:
I suggest adding the above text as a comment in the code.

Reply:
@@achouhan09, I'm not sure I understand. even if the source bucket is not created yet, don't we still want to check the replication_rules against destinations?

Reply:
Actually, here if src_bucket is not present then we cannot fetch the `src_bucket._id` which was stored in DB replication_rules, as this time src_bucket would be the destination_bucket in this validation check.
Let me know if you need more explaination

Reply:
@liranmauda comment has beed added as per your suggestion, PTAL.